jenkins-build-driver/src/main/java/org/jboss/pnc/jenkinsbuilddriver/JenkinsServerFactory.java

```diff
@@ -38,4 +42,36 @@ JenkinsServer getJenkinsServer(String url) throws BuildDriverException {
         } catch (ConfigurationParseException e) {
             throw new BuildDriverException("Cannot read configuration for " + JenkinsBuildDriver.DRIVER_ID + ".", e);        }
     }
+    
+    /**
+     * This checks if jenkins does not use option Prevent Cross Site Request Forgery exploits
+     * 
+     * @param url Jenkins instance URL with port
+     * @return
+     * @throws BuildDriverException
+     */
```

Comment:
XFactory class is responsible for instantiating X classes, this method does not fit well here.